bigtop-deploy/puppet/modules/spark/templates/spark-defaults.conf

```diff
@@ -13,7 +13,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+<% if @master_url -%>
 spark.master <%= @master_url %>
+<% else -%>
+<% if scope['deploy::roles'].include? 'spark-standalone' -%>
```

Comment:
Yep, you're right. I'm also confused as well ;)
+1 to your suggestion. That way we still keep the default to yarn strategy consistent down to config level.